lib/active_merchant/billing/gateways/credorax.rb

```diff
@@ -303,6 +303,16 @@ def add_stored_credential(post, options)
         end
       end
 
+      def recurring_properties(post, stored_credential)
+        if stored_credential[:initial_transaction]
+          post[:a9] = '1'
+        else
+          post[:a9] = '2'
+          post[:g6] = stored_credential[:network_transaction_id]
+        end
+        post
+      end
```

Comment:
💬  Comment:

you don't need to return the post object and maybe a simpler assignation

```Ruby
  def recurring_properties(post, stored_credential)
    post[:a9] = stored_credential[:initial_transaction].present? ? '1' : '2'
    post[:g6] = stored_credential[:network_transaction_id] if stored_credential[:initial_transaction].present?
  end
```

Reply:
Thank you.
Solved!

Reply:
Mentioned in Slack, but I think we need to just add `network_transaction_id` if it is present, or possibly only if it is NOT an initial transaction. I think there would never be an NTID present for an initial transaction.

Reply:
Solved!
Thanks for the help!